src/Navbar.js

```diff
@@ -85,13 +105,25 @@ const Navbar = React.createClass({
     return (
       <ComponentClass {...this.props} className={classNames(this.props.className, classes)}>
         <div className={this.props.fluid ? 'container-fluid' : 'container'}>
-          {(this.props.brand || this.props.toggleButton || this.props.toggleNavKey != null) ? this.renderHeader() : null}
-          {ValidComponentChildren.map(this.props.children, this.renderChild)}
+          {((this.props.brand || this.props.toggleButton || this.props.toggleNavKey != null) && !this.hasNavbrandChild()) ? this.renderHeader() : null}
```

Comment:
Minor cosmetic thing, but can we bind this to a variable outside the return? This is quite hard to read.